parsing/tools/src/main/java/com/apache/calcite/parsing/tools/FindDialectParsingErrors.java

```diff
@@ -0,0 +1,424 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to you under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.apache.calcite.parsing.tools;
+
+import org.apache.calcite.runtime.CalciteResource;
+import org.apache.calcite.runtime.Resources;
+import org.apache.calcite.sql.parser.SqlParseException;
+import org.apache.calcite.sql.parser.SqlParser;
+import org.apache.calcite.sql.parser.SqlParserImplFactory;
+import org.apache.calcite.sql.parser.bigquery.BigQueryParserImpl;
+import org.apache.calcite.sql.parser.defaultdialect.DefaultDialectParserImpl;
+import org.apache.calcite.sql.parser.dialect1.Dialect1ParserImpl;
+import org.apache.calcite.sql.parser.hive.HiveParserImpl;
+import org.apache.calcite.sql.parser.mysql.MySQLParserImpl;
+import org.apache.calcite.sql.parser.postgresql.PostgreSQLParserImpl;
+import org.apache.calcite.sql.parser.redshift.RedshiftParserImpl;
+
+import au.com.bytecode.opencsv.CSVReader;
+
+import com.google.gson.stream.JsonWriter;
+
+import java.io.BufferedWriter;
+import java.io.FileInputStream;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.io.OutputStreamWriter;
+import java.lang.reflect.Method;
+import java.nio.charset.StandardCharsets;
+import java.text.MessageFormat;
+import java.text.ParseException;
+import java.util.ArrayList;
+import java.util.Comparator;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Locale;
+import java.util.Map;
+import java.util.Objects;
+import java.util.stream.Collectors;
+
+/**
+ * Processes a CSV file containing queries and will output a JSON file
+ * containing the results of failing queries.
+ */
+public class FindDialectParsingErrors {
+
+  private final String inputPath;
+  private final String outputPath;
+  private final Dialect dialect;
+  private final boolean groupByErrors;
+  private final int numSampleQueries;
+  private final List<MessageFormat> errorFormats;
+
+  /**
+   *  Creates a new instance of {@code FindDialectParsingErrors}. Also populates
+   *  the errorFormats list with a {@code MessageFormat} object for each custom
+   *  defined error message in CalciteResource.
+   *
+   * @param inputPath Path to the input CSV file containing queries
```

Comment:
maybe add: Assumes CSV file is delimited by commas.

Reply:
That's a good idea, I'll add that it expects a header too since it skips the first line